### PR TITLE
DAOS-3260 placement: Fix Down Target Fallback Selection Bug

### PR DESCRIPTION
**Pull request by PetFet**
Fixed a bug that occurred when selecting rebuild targets for a
layout when most/all targets were down in the original layout.
The bug occurred because the targets that were in the layout
where not then properly discluded from future selections of
fallback targets.

In order to keep track of targets that are unavailable and have
already been considered as fallback targets a d_list_t that contains
all previously considered targets. This is used to build a bitmap
and then ignore those when calling jump consistent hashing in the
same way as before.

Signed-off-by: Peter Fetros <peter.fetros@intel.com>

### CODE REVIEW

**Comment by PetFet** [comment]
@byron-marohn 
@gnailzenh 
@wangdi1 
@NiuYawei 

Ready for review, some bug-fixes for bugs found while developing reintegration and related tests.

**Comment by PetFet** [comment]
@jolivier23 or @gnailzenh 
Should be ready for merge.

File: src/placement/jump_map.c [hunk 1]
```diff
@@ -652,10 +724,10 @@ get_object_layout(struct pl_jump_map *jmap, struct pl_obj_layout *layout,
 
 	dom_used_length = (struct pool_domain *)(root->do_targets) - (root) + 1;
 
-	D_ALLOC_ARRAY(dom_used, dom_used_length);
-	D_ALLOC_ARRAY(used_targets, ((layout->ol_nr) / 8) + 1);
+	D_ALLOC_ARRAY(dom_used, (dom_used_length / 8) + 1);
+	D_INIT_LIST_HEAD(&used_targets_list);
 
-	if (dom_used == NULL || used_targets == NULL)
+	if (dom_used == NULL)
```
Comment (jolivier23) [inline src/placement/jump_map.c]: Not really your bug, but shouldn't this be D_GOTO(out, rc = -DER_NOMEM); ?

Comment (PetFet) [reply src/placement/jump_map.c]: Thanks for spotting that,
Should be fixed now.

**Comment by PetFet** [comment]
@jolivier23 Ready for merge if changes look good.

**Pull request merged**